### PR TITLE
Base fragments

### LINKED ISSUE #716: Refactor BaseFragment functionality
* background for all fragments (backstack issue)
* configure-methods not nice and there are still issues with FAB button and toolbar
* try to close keyboard on fragment change or on show mainfragment

### PR DESCRIPTION
**Pull request by JoGir**
resolves  #716

### CODE REVIEW

File: qabelbox/src/main/java/de/qabel/qabelbox/chat/view/views/ChatOverviewFragment.kt [hunk 1]
```diff
@@ -36,12 +36,13 @@ class ChatOverviewFragment() : ChatOverview, BaseFragment(), AnkoLogger {
     override lateinit var identity: Identity
 
     override val title: String by lazy { ctx.getString(R.string.conversations) }
-    override val isFabNeeded = true
 
     val adapter = ChatOverviewAdapter({
-        contact -> presenter.handleClick(contact)
+        contact ->
+        presenter.handleClick(contact)
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/chat/view/views/ChatOverviewFragment.kt]: does this have a reason? I would even prefer less lines than more:
```KOTLIN
val adapter = ChatOverviewAdapter({ presenter.handleClick(it) }, { presenter.handleLongClick(it) })
```

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/BaseFragment.kt [hunk 2]
```diff
@@ -15,7 +17,10 @@ import org.jetbrains.anko.longToast
 import org.jetbrains.anko.onUiThread
 
 
-abstract class BaseFragment : Fragment(), QblView {
+abstract class BaseFragment(protected val mainFragment: Boolean = false,
+                            protected val showOptionsMenu: Boolean = false,
+                            protected val showFABButton: Boolean = false) : Fragment(), QblView {
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/BaseFragment.kt]: just curious ... FAButton ... FloatingActionButton ... FAB ... but why FABButton? It's like LCDDisplay 😆

File: qabelbox/src/main/java/de/qabel/qabelbox/chat/view/views/ChatOverviewFragment.kt (hunk 1 above)
Comment (JoGir) [inline qabelbox/src/main/java/de/qabel/qabelbox/chat/view/views/ChatOverviewFragment.kt]: Autoformatter, changed to one line

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/BaseFragment.kt (hunk 2 above)
Comment (JoGir) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/BaseFragment.kt]: No Idea ^^

**Pull request merged**